### PR TITLE
Imports refactoring

### PR DESCRIPTION
**Pull request by smashercosmo**
**Summary**
This PR introduces several improvements to handling import statements in the app:
1. The rest of the of absolute imports were changed to relative ones
2. Type only imports
3. Imports ordering

This PR looks massive, but 99% of changes (type only imports and imports order) are eslint autofixes.

P.S. Unfortunately eslint-plugin-import-helpers doesn't currently support ordering for type imports (https://github.com/Tibfib/eslint-plugin-import-helpers/issues/27). It would be much nicer if type imports were in the separate group.

**Test plan**
Just running current tests is enough

### CODE REVIEW

File: packages/netlify-cms-backend-bitbucket/src/API.ts [hunk 1]
```diff
@@ -653,8 +653,8 @@ export default class API {
       params: {
         pagelen: 50,
         q: oneLine`
-        source.repository.full_name = "${this.repo}" 
-        AND state = "${BitBucketPullRequestState.OPEN}" 
+        source.repository.full_name = "${this.repo}"
+        AND state = "${BitBucketPullRequestState.OPEN}"
```
Comment (smashercosmo) [inline packages/netlify-cms-backend-bitbucket/src/API.ts:657]: Oh, this is something unwanted as I remember

File: package.json [hunk 2]
```diff
@@ -148,7 +149,7 @@
     "nock": "^13.0.0",
     "node-fetch": "^2.3.0",
     "npm-run-all": "^4.1.5",
-    "prettier": "^1.19.1",
+    "prettier": "^2.2.1",
```
Comment (smashercosmo) [inline package.json:152]: Had to update Prettier, because Prettier 1 doesn't support type imports

File: .prettierrc [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {
+  "arrowParens": "avoid",
```
Comment (smashercosmo) [inline .prettierrc:2]: Prettier 2 change its defaults. So decided to stick with previous defaults in order to not introduce too many changes.

File: .eslintrc.js [hunk 4]
```diff
@@ -64,10 +65,31 @@ module.exports = {
...
+            ignore: [
+              'netlify-cms-lib-util',
+              'netlify-cms-lib-widgets',
+              'netlify-cms-backend-github',
+              'netlify-cms-backend-gitlab',
+              'netlify-cms-backend-bitbucket',
+            ],
+          },
+        ],
+        '@typescript-eslint/ban-types': [0], // TODO enable in future
```
Comment (smashercosmo) [inline .eslintrc.js:89]: This rule forbids using empty object as type. Probably good idea to enable it in future.

**Comment by smashercosmo** [comment]
Now eslint-plugin-import is used for ordering imports instead of eslint-plugin-import-helpers, because now eslint-plugin-import supports grouping for type imports 🎉🎉🎉

**Comment by smashercosmo** [comment]
hey @erezrokah I was thinking, maybe I should change this PR in a way that all of the eslint autofixes and prettier formatting will be in the separate commit? Seems like it will be much easier to review. Or I can submit this PR without autofixes and formatting at all and you can apply all of them yourself. WDYT?

**Comment by erezrokah** [comment]
> Or I can submit this PR without autofixes and formatting at all and you can apply all of them yourself. WDYT

That's a great idea. Separate PRs makes sense. 
I would even start with the Prettier PR, then do the rest of the changes.

**Comment by erezrokah** [comment]
FYI, I haven't had the chance to review since I'm looking at upgrading our dependencies to allow React 17, see https://github.com/netlify/netlify-cms/issues/5376

**Comment by smashercosmo** [comment]
This PR has been split into multiple separate PRs

**Pull request closed without merging**